### PR TITLE
build: Move benchmark generation to test/mpi/autogen.sh

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

These are only needed by the testsuite.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [ ] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: test/mpi/autogen.sh [hunk 1]
```diff
@@ -62,6 +62,14 @@ check_copy() {
     fi
 }
 
+create_benchmarks() {
+    MYDEF_BOOT=$PWD/../../modules/mydef_boot
+    export PATH=$MYDEF_BOOT/bin:$PATH
+    export PERL5LIB=$MYDEF_BOOT/lib/perl5
+    export MYDEFLIB=$MYDEF_BOOT/lib/MyDef
+    (cd bench && ./autogen.sh)
+}
```
Comment (hzhou) [inline test/mpi/autogen.sh:71]: The concern is that the user may only download the testsuite tarball and this make the autogen broken. How about protecting it with -
```
if test -d `$MYDEF_BOOT` ; then
        export PATH=$MYDEF_BOOT/bin:$PATH
        export PERL5LIB=$MYDEF_BOOT/lib/perl5
        export MYDEFLIB=$MYDEF_BOOT/lib/MyDef
fi
```
MyDef is essentially a dependency. User (like me) always can install MyDef in his system then `autogen.sh` will always work.

File: test/mpi/autogen.sh [hunk 2]
```diff
@@ -78,6 +86,9 @@ if test ! -e include/mtest_mpix.h ; then
     touch include/mtest_mpix.h
 fi
 
+# Create the benchmark tests
+create_benchmarks
```
Comment (hzhou) [inline test/mpi/autogen.sh:90]: How about rename it to "generate_benchmarks"?

**Review by hzhou — CHANGES_REQUESTED** [review]

File: test/mpi/autogen.sh (hunk 2 above)
Comment (raffenet) [reply test/mpi/autogen.sh]: Done.

File: test/mpi/autogen.sh (hunk 1 above)
Comment (raffenet) [reply test/mpi/autogen.sh:74]: Done.

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Review by hzhou — APPROVED** [review]
LGTM

**Pull request merged**